logstash-core/src/main/java/org/logstash/execution/AbstractPipelineExt.java

```diff
@@ -593,6 +583,71 @@ private UptimeMetric initOrGetUptimeMetric(final ThreadContext context,
         return retrievedMetric.toJava(UptimeMetric.class);
     }
 
+    private void initializePqFlowMetrics(final ThreadContext context, final RubySymbol[] flowNamespace, final Metric<Number> uptimeInPreciseSeconds) {
+        final IRubyObject queueContext = getSetting(context, QueueFactoryExt.QUEUE_TYPE_CONTEXT_NAME);
+        if (!queueContext.isNil() && queueContext.asJavaString().equals(QueueFactoryExt.PERSISTED_TYPE)) {
+
+            final RubySymbol[] queueNamespace = buildNamespace(QUEUE_KEY);
+            final RubySymbol[] queueCapacityNamespace = buildNamespace(QUEUE_KEY, CAPACITY_KEY);
+
+            final Supplier<NumberGauge> eventsGaugeMetricSupplier = () -> initOrGetNumberGaugeMetric(context, queueNamespace, EVENTS_KEY).orElse(null);
+            final FlowMetric growthEventsFlow = createFlowMetric(QUEUE_PERSISTED_GROWTH_EVENTS_KEY, eventsGaugeMetricSupplier, () -> uptimeInPreciseSeconds);
+            this.flowMetrics.add(growthEventsFlow);
+            storeMetric(context, flowNamespace, growthEventsFlow);
+
+            final Supplier<NumberGauge> queueSizeInBytesMetricSupplier = () -> initOrGetNumberGaugeMetric(context, queueCapacityNamespace, QUEUE_SIZE_IN_BYTES_KEY).orElse(null);
+            final FlowMetric growthBytesFlow = createFlowMetric(QUEUE_PERSISTED_GROWTH_BYTES_KEY, queueSizeInBytesMetricSupplier, () -> uptimeInPreciseSeconds);
+            this.flowMetrics.add(growthBytesFlow);
+            storeMetric(context, flowNamespace, growthBytesFlow);
+        }
+    }
+
+    private void initializePluginFlowMetrics(final ThreadContext context, final Metric<Number> uptimeInPreciseMillis) {
+        for (IRubyObject rubyObject: lirExecution.inputs()) {
+            IRubyObject rubyfiedId = rubyObject.callMethod(context, "id");
+            String id = rubyfiedId.toString();
+            initializePluginThroughputFlowMetric(context, uptimeInPreciseMillis, id);
```

Comment:
Throughput is expressed in events-per-_second_, not -per-_milli_, but lets pass our actual `UptimeMetric` through so it can get the view it needs.

```suggestion
            initializePluginThroughputFlowMetric(context, uptime, id);
```